### PR TITLE
Improve Croquemort error handling

### PR DESCRIPTION
**Pull request by noirbizarre**
This PR use `requests.ConnectTimeout` in addition to `requests.ConnectionError` in croquemort error handling to catch all raised errors including timeouts and add a logging to sentry in each case.

### CODE REVIEW

File: udata/core/dataset/croquemort.py [hunk 1]
```diff
@@ -64,8 +64,8 @@ def check_url_from_cache(url, group=None):
         response = requests.get(retrieve_url,
                                 params={'url': url, 'group': group},
                                 timeout=TIMEOUT)
-    except requests.ConnectionError:
-        return {'error': 'URL checker not found'}, {}
+    except requests.RequestException:
```
Comment (davidbgk) [inline udata/core/dataset/croquemort.py]: I think that the `ConnectionError` is more specific here and that we should keep it to make the difference between the explicit checked not found and other errors.

File: udata/core/dataset/croquemort.py [hunk 2]
```diff
@@ -85,8 +85,8 @@ def check_url_from_group(group):
         response = requests.get(retrieve_url,
                                 params={'group': group},
                                 timeout=TIMEOUT)
-    except requests.ConnectionError:
-        return {'error': 'URL checker not found'}, {}
+    except requests.RequestException:
```
Comment (davidbgk) [inline udata/core/dataset/croquemort.py]: Same here.

**Comment by davidbgk** [comment]
r+wc

File: udata/core/dataset/croquemort.py (hunk 1 above)
Comment (noirbizarre) [inline udata/core/dataset/croquemort.py]: The "not found" case is tested on the line below (`response.status_code == 404`).
As the `timeout` parameter is given as parameter, `TimeoutException` can be raised.
Am I wrong ?

File: udata/core/dataset/croquemort.py (hunk 2 above)
Comment (noirbizarre) [inline udata/core/dataset/croquemort.py]: Same here ;)

File: udata/core/dataset/croquemort.py (hunk 1 above)
Comment (davidbgk) [inline udata/core/dataset/croquemort.py]: A 404 is returned when the URL is not found on the server, not when the server is not up and running. I think that we should differentiate `ConnectionError` and `ReadTimeout` (and maybe fallback on `RequestException`?). The first one happens when there is a server error, the second when there is a network error. Having that information can help us debug the issue more fast.

Comment (noirbizarre) [inline udata/core/dataset/croquemort.py]: I know, but 404 don't raise exception and is handled just the line after

**Comment by davidbgk** [comment]
r+

**Pull request merged**